beacon-chain/powchain/service.go

```diff
@@ -243,7 +280,7 @@ func (s *Service) Start() {
 	}
 
 	// Exit early if eth1 endpoint is not set.
-	if s.currHttpEndpoint == "" {
+	if s.currHttpEndpoint == nil {
```

Comment:
this nil check would be bypassed as you have initialized it above with its zero value of the struct.

Reply:
:eyes: 
wow, good catch!